AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -105,13 +105,16 @@ + (void)setLogUrl:(NSString *)logUrl {
 }
 
 + (void)setEnabled:(BOOL)isEnabled {
-  @synchronized([MSAppCenter sharedInstance]) {
-    if ([[MSAppCenter sharedInstance] canBeUsed]) {
-      [[MSAppCenter sharedInstance] setEnabled:isEnabled];
-    }
-  }
+  [[MSAppCenter sharedInstance] setEnabled:isEnabled];
 }
 
+/**
+ * Checks if SDK is enabled and initialized.
+ *
+ * NOTE: This method is different from the instance one and in addition checks canBeUsed.
+ *
+ * @return `YES` if SDK is enabled and initialized, `NO` otherwise
+ */
 + (BOOL)isEnabled {
   @synchronized([MSAppCenter sharedInstance]) {
```

Comment:
Why is `setEnabled` not synchronized but `isEnabled` is?

Reply:
For consistency, we moved `@synchronized` calls to corresponding instance methods. So `setEnabled` is actually synchronized but in the instance method. Please check line 492.

But we left static `isEnabled` method as is because it not only calls the corresponding instance method, it also checks `canBeUsed`. So it's different from the instance `isEnabled` method and technically it's `isEnabledAndCanBeUsed`.

Unfortunately, we're not able to move `@synchronized` and `canBeUsed` to the instance method because it will break initialization of SDK — initialization method calls `isEnabled` instance method, but `canBeUsed` checks if SDK is initialized.

Please let me know if you have other ideas.